src/clj_jargon/permissions.clj

```diff
@@ -108,11 +108,10 @@
        (map #(FilePermissionEnum/valueOf (Integer/parseInt (first %))))
        (set)))
 
-(defn user-dataobject-perms
+(defn- user-dataobject-perms
   [cm user data-path]
   (validate-path-lengths data-path)
-  (->> (conj (set (user-groups cm user)) user)
-       (map (partial username->id cm))
+  (->> (conj (set (user-group-ids cm user)) (username->id cm user))
```

Comment:
Introducing this user-group-ids function was just for this spot. This way, we only need to call `username->id` once per call, rather than once for the main user plus once per group.